tests/std/tests/GH_004023_mdspan_fwd_prod_overflow/test.compile.pass.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstdint>
+#include <mdspan>
+#include <type_traits>
+
+using namespace std;
+
+struct layout_packed_upper { // LAPACK packed storage, VERY INCOMPLETE
+    template <class Extents>
+    class mapping {
+    public:
+        using extents_type = Extents;
+        using index_type   = typename extents_type::index_type;
+        using size_type    = typename extents_type::size_type;
+        using rank_type    = typename extents_type::rank_type;
+        using layout_type  = layout_packed_upper;
+
+        constexpr mapping() noexcept = default;
+        constexpr mapping(const extents_type& ex_) : ex{ex_} {}
+
+        constexpr const extents_type& extents() const {
+            return ex;
+        }
+
+        constexpr index_type required_span_size() const {
+            if (dim % 2 == 0) {
+                return (dim / 2) * (dim + 1);
+            } else {
+                return ((dim + 1) / 2) * dim;
+            }
+        }
+
+    private:
+        extents_type ex = extents_type();
+        index_type dim  = ex.extent(0);
+    };
+};
+
+int main() {
+    constexpr int32_t dim        = 47'000; // sqrt(2^32) > dim > sqrt(2^31), dim assumed even below
+    constexpr auto expected_size = [] {
+        int32_t unused;
+        bool overflow = _Mul_overflow(dim, dim, unused);
+        assert(overflow);
+
+        using UType     = remove_cv_t<make_unsigned_t<decltype(dim)>>;
+        const auto udim = static_cast<UType>(dim);
+        UType result{};
+        overflow = _Mul_overflow<UType>(udim, udim, result);

```

Comment:
The explicit template argument here is unnecessary, and the signature `(const _Int _Left, const _Int _Right, _Int& _Out)` leaves no room for doubt.

Reply:
Ah, yes, I didn't notice this cleanup. Previously I was getting an "unable to deduce template parameter" error, but that was because I lacked the `remove_cv_t` so `UType` was `const uint_32`.